cls/SourceControl/Git/PullEventHandler/IncrementalLoad.cls

```diff
@@ -44,32 +44,45 @@ Method OnPull() As %Status
 }
 
 Method DeleteFile(item As %String) As %Status
-{
-    set sc = $$$OK
-    set type = ##class(SourceControl.Git.Utils).Type(item)
-    set name = ##class(SourceControl.Git.Utils).NameWithoutExtension(item)
-    set deleted = 1
-    if type = "prj" {
-        set sc = $system.OBJ.DeleteProject(name)
-    }elseif type = "cls" {
-        set sc = $system.OBJ.Delete(item)
-    }elseif $listfind($listbuild("mac","int","inc","bas","mvb","mvi"), type) > 0 {
-        set sc = ##class(%Routine).Delete(item)
-    }elseif type = "csp" {
-        set sc = $System.CSP.DeletePage(item)
-    }elseif ##class(SourceControl.Git.Utils).UserTypeCached(item) {
-        set sc = ##class(%Library.RoutineMgr).Delete(item)
-    } else {
-        set deleted = 0
-    }
-    
-    if deleted && $$$ISOK(sc) {
-        do ##class(SourceControl.Git.Utils).RemoveRoutineTSH(item)
-        kill $$$TrackedItems(##class(SourceControl.Git.Utils).NormalizeExtension(item))
-    } else {
-        if +$system.Status.GetErrorCodes(sc) = $$$ClassDoesNotExist {
-            // if something we wanted to delete is already deleted -- good!
+{   
+    try {
+        set sc = $$$OK
+        set type = ##class(SourceControl.Git.Utils).Type(item)
+        set name = ##class(SourceControl.Git.Utils).NameWithoutExtension(item)
+        set deleted = 1
+        if type = "prj" {
+            set sc = $system.OBJ.DeleteProject(name)
+        }elseif type = "cls" {
+            set sc = $system.OBJ.Delete(item)
+        }elseif $listfind($listbuild("mac","int","inc","bas","mvb","mvi"), type) > 0 {
+            set sc = ##class(%Routine).Delete(item)
+        }elseif type = "csp" {
+            set sc = $System.CSP.DeletePage(item)
+        }elseif ##class(SourceControl.Git.Utils).UserTypeCached(item) {
+            set sc = ##class(%Library.RoutineMgr).Delete(item)
+        } else {
+            set deleted = 0
+        }
+        
+        if deleted && $$$ISOK(sc) {
+            do ##class(SourceControl.Git.Utils).RemoveRoutineTSH(item)
+            kill $$$TrackedItems(##class(SourceControl.Git.Utils).NormalizeExtension(item))
+        } else {
+            if +$system.Status.GetErrorCodes(sc) = $$$ClassDoesNotExist {
+                // if something we wanted to delete is already deleted -- good!
+                set sc = $$$OK
+            }
+        }
+    } catch e {
+        set filename = ##class(SourceControl.Git.Utils).FullExternalName(.item)
```

Comment:
One more concern - why is item byref? Does this change it? If it can then RemoveRoutineTSH behaves differently in this case.

Reply:
Going down the stack seems like SourceControl.Git.Utils:Name woudl change it, so will remove